core/classes/User.php

```diff
@@ -1087,4 +1127,26 @@ public function getUserTemplates() {
 
         return $this->_db->query('SELECT template.id, template.name FROM nl2_templates AS template WHERE template.enabled = 1 AND template.id IN (SELECT template_id FROM nl2_groups_templates WHERE can_use_template = 1 AND group_id IN ' . $groups . ')')->results();
     }
+
+    /**
+     * Save/update this users placeholders.
+     * 
+     * @param int $server_id Server ID from staffcp -> integrations to assoc these placeholders with.
+     * @param array $placeholders Key/value array of placeholders name/value from API endpoint.
+     */
+    public function savePlaceholders($server_id, $placeholders) {
+        foreach ($placeholders as $name => $value) {
+            Placeholders::getInstance()->registerPlaceholder($name);
+
+            $this->_db->query('INSERT INTO nl2_users_placeholders (server_id, uuid, name, value, last_updated) VALUES (?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE value = ?, last_updated = ?', [
+                $server_id,
+                $this->data()->uuid,
+                $name,
+                $value,
+                time(),
```

Comment:
Not sure this will work on PHP 5, might have to declare a variable with value `time()` first

Reply:
Ahh classic legacy. Thank you